src/geo/data-providers/geojson/histogram-geojson-data-provider.js

```diff
@@ -18,16 +18,16 @@ HistogramGeoJSONDataProvider.prototype.getData = function () {
     end = filter.getMax(columnName);
     start = filter.getMin(columnName);
     values = filter.getValues();
-    bins = d3.layout.histogram().bins(numberOfBins)(values.map(function (f) {
+    bins = d3.layout.histogram().range([start, end])(values.map(function (f) {
       return f.properties[options.column];
     }));
-    width = (end - start) / options.data.length;
+    width = (end - start) / bins.length;
```

Comment:
After some debugging, I feel we missed to set the range and remove the number of bins, because in the zoomed histogram the number of bins are calculated by the "provider", in this case D3 will calculate them by default with Sturges formula.